Source/WTF/wtf/NativePromise.h

```diff
@@ -593,27 +600,33 @@ class NativePromise final : public NativePromiseBase, public ConvertibleToNative
 
 public:
     template <class Dispatcher>
-    static Ref<AllPromiseType> all(Dispatcher& targetQueue, Vector<Ref<NativePromise>>& promises)
+    static Ref<AllPromiseType> all(Dispatcher& targetQueue, const Vector<Ref<NativePromise>>& promises)
     {
         static_assert(LooksLikeRCSerialDispatcher<typename RemoveSmartPointer<Dispatcher>::type>::value, "Must be used with a RefCounted SerialFunctionDispatcher");
-        if (promises.isEmpty())
-            return AllPromiseType::createAndResolve(Vector<ResolveValueType>());
-
+        if (promises.isEmpty()) {
+            if constexpr (std::is_void_v<ResolveValueT>)
+                return AllPromiseType::createAndResolve();
+            else
+                return AllPromiseType::createAndResolve(typename AllPromiseType::ResolveValueType());
+        }
         auto producer = adoptRef(new AllPromiseProducer(promises.size()));
         auto promise = producer->promise();
         for (size_t i = 0; i < promises.size(); ++i) {
             promises[i]->whenSettled(targetQueue, [producer, i] (ResultParam result) {
-                if (result)
-                    producer->resolve(i, maybeMove(result.value()));
-                else
+                if (result) {
+                    if constexpr (std::is_void_v<ResolveValueT>)
+                        producer->resolve(i, std::nullptr_t());
```

Comment:
I was not sure why you were using `nullptr_t()` here, I had to look at `resolve()` to understand that it's not actually used, so it could be anything.
To reduce this potential confusion, I would suggest you use the same `std::monostate` (or other empty struct as suggested above), it will be easier to make the connection with your void-type-alternative.

Reply:
yeah, I only used it to avoid duplicating code as I needed an argument, any argument.